uf-mil/mil
# 1299 alexj boat drone comms

## Description

This PR creates a ROSSerialDevice called DroneCommDevice which allows Navigator to communicate with the drone. Also, it adds a test-suite for simulated packets.


## Screenshot or Video

<img width="473" alt="Screenshot 2024-10-30 at 7 25 36 PM" src="https://github.com/user-attachments/assets/3bec9c50-a9ac-4ba2-93c0-7693afcf411e">
<img width="865" alt="Screenshot 2024-10-30 at 11 57 08 PM" src="https://github.com/user-attachments/assets/ddfd2dd7-747f-4acf-b625-e6d4ce4fa14c">



## Related Issues


\- #1299 

## Testing


rosrun navigator_drone_comm test_simulated_drone.py
- This starts the test program.
- You should see a received GPSDronePacket and TargetPacket printed to the console immediately.

While "Testing receiving heartbeats for 5 secs..."
- You should see no outputted errors for missing heartbeat.

While the long test loop runs
- You should see error for no received heartbeat every second.
- Use ctrl Z to end execution.

To test services that send messages:
- Comment out lines in driver.py with “send_packet()”.
- Uncomment the print statements next to those lines.
- Now instead of freezing after sending packets, you will see print statements when messages would be sent to the drone.

Now, while the long test runs:
- Call the services below in a separate terminal to check that a message would be sent.
- Use ctrl c to end execution early.

rosservice call /test_simulated_drone/start “sample_mission”
rosservice call /test_simulated_drone/estop
rosservice call /test_simulated_drone/stop

## About This PR


- [X] I have updated documentation related to this change so that future members are aware of the changes I've made.
